build.gradle

```diff
@@ -707,8 +707,35 @@ class JDKDetails {
     }
 
     String createDownloadUrl() {
+        return createElasticCatalogDownloadUrl()
+    }
+
+    private String createElasticCatalogDownloadUrl() {
+        // Ask details to catalog https://jvm-catalog.elastic.co/jdk/latest_adoptiumjdk_17_windows
+        // and return the url to download the JDK
+
+        // arch x86_64 never used, only aarch64 if macos
+        def url = "https://jvm-catalog.elastic.co/jdk/latest_adoptiumjdk_${major}_${osName}"
+
+        // Append the cpu's arch only if Mac on aarch64, all the other OSes doesn't have CPU extension
```

Comment:
Hm, I don't think that's correct, for example for Linux `aarch64` artifacts we have https://jvm-catalog.elastic.co/jdk/latest_adoptiumjdk_17_linux_aarch64, for Windows e.g. `/jdk/latest_zulu_19_windows_aarch64` or even 32bit variants like `/jdk/latest_adoptopenjdk_16_windows_x86_32`.

Reply:
I can't use a generalized URL template like `https://jvm-catalog.elastic.co/jdk/latest_adoptiumjdk_${major}_${osName}_${arch}` because the url https://jvm-catalog.elastic.co/jdk/latest_adoptiumjdk_17_linux_x86_64 return a 
```
Could not find download for latest_adoptiumjdk_17_linux_x86_64
```

But the https://jvm-catalog.elastic.co/jdk/latest_adoptiumjdk_17_linux return the `x86_64` artifact.

So we could use the arch format iff it's an ARM (not only for MacOS but also for Linux).

WDYT?

Reply:
Fixed by b96888828b684c3f31d95893d671a188bae3553a